File: server/src/Tables/Pill.cs
using SpacetimeDB;

namespace pillz.server.Tables;

[Table(Public = true)]
[SpacetimeDB.Index.BTree(Name = "PlayerId", Columns = [nameof(PlayerId)])]
public partial struct Pill
{
    [PrimaryKey]
    public uint EntityId;
    public uint PlayerId;
    public DbVector2 Direction;
    public DbVector2 Position;
    public int Hp;
    public int Dmg;
    public uint Frags;
    public Jetpack Jetpack;
    public DbVector2 AimDir;
    public DbVector2? Force;
    public WeaponType SelectedWeapon;
    public Weapon PrimaryWeapon;
    public Weapon SecondaryWeapon;
}

File: server/src/tables/pill.rs
use crate::math::dbvector2::DbVector2;
use crate::types::jetpack::Jetpack;
use crate::types::weapon::Weapon;
use crate::types::weapon_type::WeaponType;

#[spacetimedb::table(name = pill, public)]
#[derive(Debug, Clone)]
pub struct Pill {
    #[primary_key]
    pub entity_id: u32,
    #[index(btree)]
    pub player_id: u32,
    pub direction: DbVector2,
    pub position: DbVector2,
    pub hp: i32,
    pub dmg: i32,
    pub frags: u32,
    pub jetpack: Jetpack,
    pub aim_dir: DbVector2,
    pub force: Option<DbVector2>,
    pub selected_weapon: WeaponType,
    pub primary_weapon: Weapon,
    pub secondary_weapon: Weapon,
    pub stims: i32,
}
